MVC-App/Repositories/UnitOfWork.cs

```diff
@@ -0,0 +1,93 @@
+﻿using MVC_App.Domain.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace MVC_App.Repositories
+{
+    public class UnitOfWork : IDisposable
+    {
+        private RelationContext context = new RelationContext();
+
+        private GenericRepository<Relation> relationRepository;
```

Comment:
Please create RelationRepository
...
    public class RelationRepository : GenericRepository<Relation>, IRepository<Relation>
    {
    }
...
Then youe line 14 will be private RelationRepository relationRepository;

Reply:
The same for the others repos.